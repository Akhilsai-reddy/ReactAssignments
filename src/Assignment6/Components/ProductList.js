import React from "react";
import Products from "./Products";

function ProductsList(){
  

    return (
      <>
        <table border={2}>
          <thead>
            <tr>
              <th>PRODUCT NAME</th>
              <th>QUANTITY</th>
              <th>PRICE</th>
            </tr>
          </thead>
          <tbody>
            <Products/>
          </tbody>
        </table>
      </>
    );
  
}

export default ProductsList;
