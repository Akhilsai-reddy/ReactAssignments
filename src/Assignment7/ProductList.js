import React from 'react'
import Product from './Product'

const ProductList = (props) => {

  return (
    <>
     <table border={2}>
        <thead>
          <tr>
            <th>ID</th>
            <th>PRODUCT NAME</th>
            <th>QUANTITY</th>
            <th>PRICE</th>
          </tr>
        </thead>
        <tbody>
        {
            props.Products.map((product,index)=>
            <Product key={index} ID={product.ID} Name={product.Product_Name} Quantity={product.Quantity} Price={product.Price}/>)
        }
        </tbody>
     </table>
    </>
  )
}

export default ProductList