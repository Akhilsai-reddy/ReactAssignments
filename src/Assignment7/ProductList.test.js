import ProductList from "./ProductList";
import { mount} from "enzyme";

describe('rendering ProductsList',()=>{
    let wrapper
    let props

    beforeEach(()=>{
        props={
            Products:[{
                "ID": 1,
                "Price": "$200",
                "Product_Name": "MOTO",
                "Quantity": 3
            },
            {
                "ID": 2,
                "Price": "$290",
                "Product_Name": "REDMI",
                "Quantity": 4
            },
            {
                "ID": 3,
                "Price": "$260",
                "Product_Name": "SONY",
                "Quantity": 6
            }
        ]
        }

        wrapper=mount(<ProductList Products={props.Products}/>)
    })
    it('render the length of the th',()=>{
        expect(wrapper.find('th').length).toEqual(4)
    })

    it('renders 1st th as ID',()=>{
        expect(wrapper.find('th').at(0).text()).toBe("ID")
    })

    it('renders 3rd th as Quantity',()=>{
        expect(wrapper.find('th').at(2).text()).toEqual("QUANTITY")
    })

    it('rendering ProductList',()=>{
        expect(wrapper).toMatchSnapshot();
    })

})