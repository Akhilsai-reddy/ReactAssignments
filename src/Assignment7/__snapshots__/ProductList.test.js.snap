// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering ProductsList rendering ProductList 1`] = `
<ProductList
  Products={
    Array [
      Object {
        "ID": 1,
        "Price": "$200",
        "Product_Name": "MOTO",
        "Quantity": 3,
      },
      Object {
        "ID": 2,
        "Price": "$290",
        "Product_Name": "REDMI",
        "Quantity": 4,
      },
      Object {
        "ID": 3,
        "Price": "$260",
        "Product_Name": "SONY",
        "Quantity": 6,
      },
    ]
  }
>
  <table
    border={2}
  >
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          PRODUCT NAME
        </th>
        <th>
          QUANTITY
        </th>
        <th>
          PRICE
        </th>
      </tr>
    </thead>
    <tbody>
      <Product
        ID={1}
        Name="MOTO"
        Price="$200"
        Quantity={3}
        key="0"
      >
        <tr>
          <td>
            1
          </td>
          <td>
            MOTO
          </td>
          <td>
            3
          </td>
          <td>
            $200
          </td>
        </tr>
      </Product>
      <Product
        ID={2}
        Name="REDMI"
        Price="$290"
        Quantity={4}
        key="1"
      >
        <tr>
          <td>
            2
          </td>
          <td>
            REDMI
          </td>
          <td>
            4
          </td>
          <td>
            $290
          </td>
        </tr>
      </Product>
      <Product
        ID={3}
        Name="SONY"
        Price="$260"
        Quantity={6}
        key="2"
      >
        <tr>
          <td>
            3
          </td>
          <td>
            SONY
          </td>
          <td>
            6
          </td>
          <td>
            $260
          </td>
        </tr>
      </Product>
    </tbody>
  </table>
</ProductList>
`;
